packages/components/src/mobile/link-settings/test/edit.native.js

```diff
@@ -34,9 +47,7 @@ describe.each( [
 			type: 'core/button',
 			initialHtml: `
 				<!-- wp:button {"style":{"border":{"radius":"5px"}}} -->
-				<div class="wp-block-button">
-					<a class="wp-block-button__link" style="border-radius:5px">Link</a>
-				</div>
+				<div class="wp-block-button"><a class="wp-block-button__link wp-element-button" style="border-radius:5px">Link</a></div>
```

Comment:
I suppose this is a good example of the benefit of leveraging UI instead of initial HTML. I wonder if there is are benefits to relying upon initial HTML. Test speed possibly? I'm unsure of if the impact is negligible.

Reply:
I think the cases where we use initial HTML are mainly driven to start the editor in a specific state, similar to opening a post with content. This has the downside of using fixed versions of the block's HTML code, which as in this case, it would eventually lead to being outdated.

It's likely that adding the block and setting the content is slower than using the initial HTML. I haven't reviewed thoroughly these test cases to determine if could use a different approach, my main goal in this PR was to simply fix the issues to unblock the RN upgrade. Nevertheless, we could improve this in another PR if we feel the need to.

Reply:
> I haven't reviewed thoroughly these test cases to determine if could use a different approach, my main goal in this PR was to simply fix the issues to unblock the RN upgrade. Nevertheless, we could improve this in another PR if we feel the need to.

Definitely. No intention from my side to modify the testing approach in this PR. Merely shared the thought for general consideration.